### ISSUE TITLE
Enhancement: `warnOnUnsupportedTypeScriptVersion` with a specific TypeScript version

### ISSUE THREAD

**Issue by Peeja** [typescript-eslint/typescript-eslint#7128]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

When using an unsupported TypeScript version, the parser shows a warning message. As described in #6934, the temporary solution to this (when the warning is read and deemed to be not an issue, as it is for TS 5.1) is to set `warnOnUnsupportedTypeScriptVersion` to `false`.

However, config like this is easy to forget about. In the future, if this happens again, it's likely most people who turned on `warnOnUnsupportedTypeScriptVersion` will still have it on, and therefore not see the warning (which may be more significant the next time it comes up).

I'd like to propose that the `warnOnUnsupportedTypeScriptVersion` be enhanced to accept a TypeScript version, rather than simply a boolean. Given a TS version number, it would then only warn for newer versions of TypeScript. The idea, then, is that once you upgrade TS in your project, the warning becomes automatically re-enabled. Or, to put it another way, rather than tell the parser that you're comfortable with all unsupported TS versions until the end of time, you only need to tell it that you're comfortable with *this* unsupported TS version, because you've looked into it and determined that it's not a problem.

One wrinkle, though, is that `warnOnUnsupportedTypeScriptVersion: "5.1.3"` actually reads more like the opposite of what it would mean. Options include making a different option for it with a different name, or accepting a version spec so that you could write `warnOnUnsupportedTypeScriptVersion: ">5.1.3"`.

### Fail

```typescript
// No code; this is just config
```

### Pass

```typescript
// No code; this is just config
```

**Comment by bradzacher** [comment]
Honestly the advice to disable the warning isn't good because really we do not support 5.1 because there are known increased chances for OOMs on even small projects due to a regression within TS - which is the whole reason we haven't been able to land support yet.

People can continue to use our tooling at their own risk until we officially support it, but disabling the warning is folly IMO as it lulls people into a false sense of security.

Which is exactly why you want an option to disable it for a specific version - so you don't forget to turn it off and can be warned in future as a way of saying "I accept the risks for this version, don't warn me".

But the thing we don't want is for some to race ahead of us, turn off the warning, then forget they turned it off and report issues caused by a version we don't yet support.

I don't know if allowing that is a good idea TBH.

**Comment by bradzacher** [comment]
Closing as we've released support for 5.1 and I don't think there's enough community interest in this for now.

**Comment by Peeja** [comment]
Even better! 😀
